tests/test_toolchain.py

```diff
@@ -0,0 +1,62 @@
+import sys
+import pytest
+import mock
+from pythonforandroid.toolchain import ToolchainCL
+from pythonforandroid.util import BuildInterruptingException
+
+
+class TestToolchainCL:
+
+    def test_help(self):
+        """
+        Calling with `--help` should print help and exit 0.
+        """
+        argv = ['toolchain.py', '--help', '--storage-dir=/tmp']
+        with mock.patch('sys.argv', argv), pytest.raises(SystemExit) as ex_info, \
+                mock.patch('argparse.ArgumentParser.print_help') as m_print_help:
+            ToolchainCL()
+        assert ex_info.value.code == 0
+        assert m_print_help.call_args_list == [mock.call()]
+
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="requires python3")
+    def test_unknown(self):
+        """
+        Calling with unknown args should print help and exit 1.
+        """
+        argv = ['toolchain.py', '--unknown']
+        with mock.patch('sys.argv', argv), pytest.raises(SystemExit) as ex_info, \
+                mock.patch('argparse.ArgumentParser.print_help') as m_print_help:
+            ToolchainCL()
+        assert ex_info.value.code == 1
+        assert m_print_help.call_args_list == [mock.call()]
+
+    def test_create(self):
+        """
+        Basic `create` distribution test.
+        """
+        argv = ['toolchain.py', 'create', '--sdk-dir=/tmp/android-sdk', '--ndk-dir=/tmp/android-ndk']
```

Comment:
About the tox error I got, mentioned in [this comment](https://github.com/kivy/python-for-android/pull/1928#issuecomment-514932867), the addition of `dist_name` in argv solves the problem:
```
        argv = [
            'toolchain.py',
            'create',
            '--sdk-dir=/tmp/android-sdk',
            '--ndk-dir=/tmp/android-ndk',
            '--dist-name=test_toolchain',
        ]
```

Reply:
Strangely enough I just tested it and it runs through fine for me, locally. I tested with python 3.7.3 and python 2.7.16 on Fedora 30. Or was this already fixed?

Reply:
No I haven't applied the fix yet. That's a very strange behaviour indeed.
Thank you @opacam for sharing a fix. I'll look into it

Reply:
I've added it, build is still passing for me, let's see for Travis and for you then :smile:

Reply:
Yeah, now works, for CI and for my local tox tests too :smile:

Thanks @AndreMiras!!